src/components/PostsList.tsx

```diff
@@ -1,85 +1,80 @@
+import classNames from 'classnames';
 import React from 'react';
 
-export const PostsList: React.FC = () => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+import { Post } from '../types/Post';
 
-    <table className="table is-fullwidth is-striped is-hoverable is-narrow">
-      <thead>
-        <tr className="has-background-link-light">
-          <th>#</th>
-          <th>Title</th>
-          <th> </th>
-        </tr>
-      </thead>
+type Props = {
+  posts: Post[];
+  selectedPostId: number;
+  onPost: (postDetail: Post | null) => void;
+};
 
-      <tbody>
-        <tr data-cy="Post">
-          <td data-cy="PostId">17</td>
+export const PostsList: React.FC<Props> = ({
+  posts,
+  selectedPostId,
+  onPost,
+}) => {
+  const onClickHandle = (post: Post) => {
+    onPost(selectedPostId === post.id ? null : post);
+  };
 
-          <td data-cy="PostTitle">
-            fugit voluptas sed molestias voluptatem provident
-          </td>
+  return (
+    <>
+      {!posts.length
+        ? (
+          <div
+            className="notification is-warning"
+            data-cy="NoPostsYet"
+          >
+            No posts yet
+          </div>
+        )
+        : (
```

Comment:
you can use early return if there are no posts
```suggestion
  if (!posts.length) {
    return (
      <div className="notification is-warning" data-cy="NoPostsYet">
        No posts yet
      </div>
    )
  }
  
  return (
```